cmd/server-main.go

```diff
@@ -250,20 +250,26 @@ func serverMain(ctx *cli.Context) {
 	// are intentionally providing backward compatibility.
 	{
 		// Check for backward compatibility and newer style.
-		if !globalIsEnvCreds && globalIsDistXL {
+		if !globalIsEnvCreds {
 			// Try to load old config file if any, for backward compatibility.
-			var config = &serverConfig{}
+			var config = &serverConfigV33{}
 			if _, err = Load(getConfigFile(), config); err == nil {
-				globalActiveCred = config.Credential
+				if config.Credential.IsValid() {
+					globalIsEnvCreds = true
+					globalEnvCred = config.Credential
+				}
 			}
 
 			if os.IsNotExist(err) {
 				if _, err = Load(getConfigFile()+".deprecated", config); err == nil {
-					globalActiveCred = config.Credential
+					if config.Credential.IsValid() {
+						globalIsEnvCreds = true
+						globalEnvCred = config.Credential
+					}
 				}
 			}
 
-			if globalActiveCred.IsValid() {
+			if globalEnvCred.IsValid() {
```

Comment:
on L278 - That is an in-appropriate error for an fs mode. For an fs it should be - `Unable to initialize the server in fs mode`

Reply:
This should be just for dist XL

Reply:
what i meant is - if we unset the access/secret ket env and try to start in fs mode - we get this `Unable to initialize the server in distributed mode: Credentials missing.` which is misleading

Reply:
No, I understand, I added a bug here - brought back the code from server.